python_modules/dagster/dagster/_core/definitions/remote_asset_graph.py

```diff
@@ -345,11 +347,14 @@ def all_job_names(self) -> AbstractSet[str]:
         return {job_name for node in self.asset_nodes for job_name in node.job_names}
 
     @cached_property
-    def repository_handles_by_key(self) -> Mapping[AssetKey, RepositoryHandle]:
-        return {k: node.priority_repository_handle for k, node in self._asset_nodes_by_key.items()}
+    def repository_handles_by_key(self) -> Mapping[EntityKey, RepositoryHandle]:
+        return {
+            **{k: node.priority_repository_handle for k, node in self._asset_nodes_by_key.items()},
+            **self._repository_handles_by_asset_check_key,
+        }
```

Comment:
ya having to key in to nodes and handles is a bit scuffed - increasingly liking the idea of a new obj